src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -82,8 +95,21 @@ func (d *downsamplerAndWriter) Write(
 	tags models.Tags,
 	datapoints ts.Datapoints,
 	unit xtime.Unit,
+	overrides MappingAndStoragePoliciesOverrides,
 ) error {
-	if d.downsampler != nil {
+	var (
```

Comment:
As discussed, perhaps split this method into two submethods to make it more clean and subdivide the complexity into smaller pieces.